project_leverage/controllers/main.py

```diff
@@ -0,0 +1,109 @@
+        month_list = []
+        month_name_list = []
+        for rec in raw_data:
+            if str(rec.get('month')).strip() + ' ' + str(
+                    int(rec.get('year'))) not in month_name_list:
+                month_name_list.append(
+                    str(rec.get('month')).strip() + ' ' + str(
+                        int(rec.get('year'))))
+            if str(rec.get('date_mm')).strip() + str(
+                    int(rec.get('year'))) not in month_list:
+                month_list.append(
+                    str(rec.get('date_mm')).strip() + str(
+                        int(rec.get('year'))))
+            if project_data.get(rec.get('employee_id') or 0):
+                project_data[rec.get('employee_id') or 0].update(
+                    {str(rec.get('date_mm')).strip() + str(
+                        int(rec.get('year'))): {'name':
+                                                employee_obj.browse(int(
+                                                    rec.get(
+                                                        'employee_id'))).name,
+                                                'month_name': rec.get('month'),
+                                                'hours': rec.get('sum'),
+                                                }})
+            else:
+                project_data.update({rec.get('employee_id') or 0: {
+                    str(rec.get('date_mm')).strip() + str(
+                        int(rec.get('year'))): {'month_name': rec.get('month'),
+                                                'hours': rec.get('sum'),
+                                                },
+                    'name': rec.get('employee_id') and employee_obj.browse(
+                        int(rec.get('employee_id'))).name or "undefine"
+                }, })
+        values['month_name_list'] = month_name_list
+        values['month_list'] = month_list
+        values['leverage'] = project_data
+        return values
+
+    def select_line(self):
+        select_str = """
+            Select to_char(date , 'Month') as month, sum(unit_amount),
+            EXTRACT(YEAR FROM date) AS Year, to_char(date, 'MM') as date_mm,
+            employee_id
+        """
+        return select_str
+
+    def line_from(self):
+        return """
+            FROM account_analytic_line
+        """
+
+    def line_where(self, project_ids):
+        project_mananger_ids = [
+            project.user_id.id
+            for project in request.env['project.project'].browse(project_ids)]
+        employee_ids = request.env['hr.employee'].search(
+            [('user_id', 'in', project_mananger_ids),
+             ('user_id', '!=', False)])
+        query = """
+            WHERE
+        """
+        if not employee_ids:
+            raise UserError('Project Manager is not defined or no '
+                            'timesheet entry found for project manager.')
+        if len(project_ids) > 1 and len(employee_ids) > 1:
```

Comment:
To check for nonzero length, this is the Pythonic way
```suggestion
        if project_ids and employee_ids:
```